stateMachine.go

```diff
@@ -0,0 +1,137 @@
+/*
+HSM provides the framework for Hierarchical State Machine implementations.
```

Comment:
I think this is a really good writeup. It would probably be useful to move into the README, instead of here, so that new users of the package are able to quickly see how things are laid out.
Also, since this is an open source project I'd avoid including links to our internal confluence page; if we can summarize the essentials either in the README or in a dependent package that will make it easier for others to use.

Reply:
Good, I'll remove the internal links and external links and add the object model as a png.

From my reading up on go best practices I learned that I should add comments like these in one file associated with the package for godoc rendering.  I also like the README.md pattern.  For now I'll copy the content into both locations.

Reply:
Makes sense! 👍